### PR TITLE
Naïve process walker for Darwin

### PR DESCRIPTION
**Pull request by peterbourgon**
This fixes the regression where process names weren't appearing for Darwin probes. Makes testing easier.

### CODE REVIEW

File: probe/main.go [hunk 1]
```diff
@@ -83,24 +83,22 @@ func main() {
 	)
 
 	// TODO provide an alternate implementation for Darwin.
```
Comment (tomwilkie) [inline probe/main.go]: Can this todo go away now?

File: probe/main.go [hunk 2]
```diff
@@ -83,24 +83,22 @@ func main() {
...
-			if err = report.AddLocalBridge(*dockerBridge); err != nil {
-				log.Fatalf("failed to get docker bridge address: %v", err)
-			}
-
-			dockerRegistry, err := docker.NewRegistry(*dockerInterval)
-			if err != nil {
-				log.Fatalf("failed to start docker registry: %v", err)
-			}
-			defer dockerRegistry.Stop()
+	if runtime.GOOS == linux && *dockerEnabled {
```
Comment (tomwilkie) [inline probe/main.go]: Alternatively, we could default dockerEnabled to false, and set it in the docker init scripts.  Then we could remove this if.

File: probe/main.go [hunk 3]
```diff
@@ -83,24 +83,22 @@ func main() {
...
+		if err = report.AddLocalBridge(*dockerBridge); err != nil {
```
Comment (tomwilkie) [inline probe/main.go]: should probably be :=

File: probe/main.go (hunk 1 above)
Comment (tomwilkie) [inline probe/main.go]: Also, the if processCache != nil line below can go too (I can't seem to comment on that line, sorry)

File: probe/process/walker_darwin.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+func (walker) Walk(f func(*Process)) error {
+	// Copied from procspy spy_darwin
+	out, err := exec.Command(
+		lsofBinary,
+		"-i",       // only Internet files
+		"-n", "-P", // no number resolving
+		"-w",             // no warnings
+		"-F", lsofFields, // \n based output of only the fields we want.
+	).CombinedOutput()
```
Comment (tomwilkie) [inline probe/process/walker_darwin.go]: :-( thats a bit icky.  Only other way I could find involves cgo:

https://github.com/jondot/gosigar/blob/master/sigar_darwin.go#L195

File: probe/main.go (hunk 2 above)
Comment (peterbourgon) [inline probe/main.go]: I don't understand what you mean by "docker init scripts"?

**Comment by peterbourgon** [comment]
OK, I think I addressed your feedback...

File: probe/main.go (hunk 2 above)
Comment (tomwilkie) [inline probe/main.go]: Sorry wasn't clear.  Things like entrypoint.sh or even our scope script.  I'm assuming you don't use them when launching on Darwin.

Comment (peterbourgon) [inline probe/main.go]: Ah, yeah, but we'd still need to check the flag here, no?

Comment (tomwilkie) [inline probe/main.go]: Yeah.  Not a big deal.  Just seems weird that we wouldn't run the docker integration and we won't error even if explicitly run with --docker=true

Comment (peterbourgon) [inline probe/main.go]: Yep, I agree with you. Updated.

**Comment by peterbourgon** [comment]
Good?

**Comment by tomwilkie** [comment]
Yeah LGTM.  Please squash, and hold until David's done the release.

**Comment by peterbourgon** [comment]
Of course.

**Pull request merged**